saleor/graphql/product/types/channels.py

```diff
@@ -24,14 +35,34 @@ class Meta:
     def resolve_channel(root: models.ProductChannelListing, info, **_kwargs):
         return ChannelByProductChannelListingIDLoader(info.context).load(root.id)
 
+    @staticmethod
+    @permission_required(ProductPermissions.MANAGE_PRODUCTS)
+    def resolve_purchase_cost(root: models.ProductChannelListing, *_args):
+        purchase_cost, _ = get_product_costs_data(root, root.channel.slug)
+        return purchase_cost
+
+    @staticmethod
+    @permission_required(ProductPermissions.MANAGE_PRODUCTS)
+    def resolve_margin(root: models.ProductChannelListing, *_args):
+        _, margin = get_product_costs_data(root, root.channel.slug)
+        return Margin(margin[0], margin[1])
+
 
 class ProductVariantChannelListing(CountableDjangoObjectType):
+    cost_price = graphene.Field(Money, description="Cost price of the variant.")
+    margin = graphene.Int(description="Gross margin percentage value.")
+
     class Meta:
         description = "Represents product varaint channel listing."
         model = models.ProductVariantChannelListing
         interfaces = [graphene.relay.Node]
-        only_fields = ["id", "channel", "price"]
+        only_fields = ["id", "channel", "price", "cost_price"]
 
     @staticmethod
     def resolve_channel(root: models.ProductVariantChannelListing, info, **_kwargs):
         return ChannelByProductVariantChannelListingIDLoader(info.context).load(root.id)
+
+    @staticmethod
+    @permission_required(ProductPermissions.MANAGE_PRODUCTS)
+    def resolve_margin(root: models.ProductVariantChannelListing, *_args):
+        return resolve_margin(root, root.channel.slug)
```

Comment:
Do we need `resolve_margin` as a separate function? Is it also used in some other place?

Reply:
No, its look more clear for me :)

Reply:
`resolvers.py` is usually used by top-level resolvers from `schema.py` and a field resolver is rather unexpected there.

Reply:
i will change that